tests/functional/mesh/mesh_test.go

```diff
@@ -94,6 +99,56 @@ func TestMeshConnections(t *testing.T) {
 			if connectionsReady == false {
 				t.Error("Timed out while waiting for connections")
 			}
+			mesh.Shutdown()
+		})
+	}
+}
+
+// Test that a mesh starts and that connections are what we expect
+func TestMeshShutdown(t *testing.T) {
+	testTable := []struct {
+		filename string
+	}{
+		{"mesh-definitions/random-mesh-tcp.yaml"},
+		{"mesh-definitions/random-mesh-udp.yaml"},
+		{"mesh-definitions/random-mesh-ws.yaml"},
+	}
+	for _, data := range testTable {
+		filename := data.filename
+		t.Run(filename, func(t *testing.T) {
+			mesh, err := NewMeshFromFile(filename)
+			if err != nil {
+				t.Error(err)
+			}
+			err = mesh.WaitForReady(10000)
+			if err != nil {
+				t.Error(err)
+			}
+			mesh.Shutdown()
+			mesh.WaitForShutdown()
+
+			// Check that the connections are closed
+			pid := os.Getpid()
+			pidString := strconv.Itoa(pid)
+			done := false
+			for timeout := 1000000; timeout > 0 && !done; {
+				var out bytes.Buffer
+				cmd := exec.Command("netstat", "-tunp")
```

Comment:
This is pretty hacky and im not a huge fan but i think it's fine

Reply:
I'm not keen on shelling out to an external command here either, but if we're going to do that, it should probably be `ss` from the iproute package.  The net-tools package has been considered deprecated for several years and it might be problematic to rely on it existing in CI environments etc.